<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Exercise sheet 7: BLAT</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>






<link rel="stylesheet" href="bioinf-style.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">



<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div style="background:white">
  </div>
  <div class="container">
    <div class="navbar-header">
      <!-- NOTE: add "navbar-inverse" class for an alternate navbar background -->
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand navbar-img" href="#">
        <div class="d-flex align-items-center h-25">
          <img src="figures/bioinf-fr-logo-blau.svg" class="d-flex align-items-center" alt="">
        </div>

      </a>
              <a class="navbar-brand">Bioinformatics II</a>

    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li><a href="index.html">Intro</a></li>
        <li><a href="howto.html">HowTo</a></li>
        <li><a href="exercise-sheet-1.html">Sheet 1</a></li>
        <li><a href="exercise-sheet-2.html">Sheet 2</a></li>
        <li><a href="exercise-sheet-3.html">Sheet 3</a></li>
        <li><a href="exercise-sheet-4.html">Sheet 4</a></li>
        <li><a href="exercise-sheet-5.html">Sheet 5</a></li>
        <li><a href="exercise-sheet-6.html">Sheet 6</a></li>
        <li><a href="exercise-sheet-7.html">Sheet 7</a></li>
        <li><a href="exercise-sheet-8.html">Sheet 8</a></li>
        <li><a href="exercise-sheet-9.html">Sheet 9</a></li>
        <!--<li><a href="exercise-sheet-10.html">Sheet 10</a></li>-->
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li><a href="http://www.bioinf.uni-freiburg.de/index.html?en2">Contact</a></li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Exercise sheet 7: BLAT</h1>

</div>


<hr />
<p>In the first step of the BLAT (Blast-like Alignment Tool) algorithm, regions that are likely to be homologous are detected. In this exercise sheet, we will investigate the search stage of the BLAT algorithm on the example of the mouse genome.</p>
<p>We want to determine whether a region from the human genome aligns to a part of the mouse genome. Therefore, a homologous region to our query sequence will be detected during the search stage.</p>
<p>We assume that:</p>
<ul>
<li>the human genome <span class="math inline">\(G\)</span> is approximately <span class="math inline">\(2.9\)</span> billion bases long.</li>
<li>the mouse genome <span class="math inline">\(G\)</span> is approximately <span class="math inline">\(2.5\)</span> billion bases long.</li>
<li>the match ratio <span class="math inline">\(M\)</span> between homologous areas of both species is <span class="math inline">\(98\%\)</span> for DNA and <span class="math inline">\(89\%\)</span> for protein alignments.</li>
<li>in our example we assume that homologous areas <span class="math inline">\(H\)</span> are typically <span class="math inline">\(50\)</span> bases long.</li>
<li>our query sequence <span class="math inline">\(Q\)</span> is GTCCTCGGAACCAGGACCTCGGCGTGGCCTAGCG.</li>
</ul>
<div id="exercise-1" class="section level1">
<h1>Exercise 1</h1>
<p>For the <span class="math inline">\(K\)</span>-mer sizes <span class="math inline">\(K=7\)</span> and <span class="math inline">\(K=14\)</span>, respectively.</p>
<div id="section" class="section level4 tabset">
<h4></h4>
<div id="question-1a" class="section level5">
<h5>Question 1A</h5>
<div class="question">
<p>What is the probability of having a perfect match between a specific <span class="math inline">\(K\)</span>-mer in a homologous region and a <span class="math inline">\(K\)</span>-mer in the query sequence?</p>
</div>
</div>
<div id="formula" class="section level5">
<h5>Formula</h5>
<div class="answer">
<p><span class="math display">\[
p_{1} = M^K
\]</span></p>
</div>
</div>
<div id="solution" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>For <span class="math inline">\(K=7\)</span>: <span class="math display">\[
p_{1} = 0.98^{7} = 0.8681
\]</span> For <span class="math inline">\(K=14\)</span>: <span class="math display">\[
p_{1} = 0.98^{14} = 0.7536
\]</span></p>
</div>
</div>
</div>
<div id="section-1" class="section level4 unnumbered">
<h4></h4>
</div>
<div id="section-2" class="section level4 tabset">
<h4></h4>
<div id="question-1b" class="section level5">
<h5>Question 1B</h5>
<div class="question">
<p>What is the probability that at least one non-overlapping <span class="math inline">\(K\)</span>-mer in the homologous region matches perfectly with the corresponding <span class="math inline">\(K\)</span>-mer in the query sequence?</p>
</div>
</div>
<div id="formula-1" class="section level5">
<h5>Formula</h5>
<div class="answer">
<p><span class="math display">\[
P = 1 - (1 - p_{1})^{T} = 1 - (1 - M^{K})^{T},\\
\text{with}\quad T= \left\lfloor\frac{H}{K}\right\rfloor
\]</span></p>
</div>
</div>
<div id="solution-1" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>For <span class="math inline">\(K=7\)</span>: <span class="math display">\[
P = 1 - (1 - 0.98^7)^{\left\lfloor\frac{50}{7}\right\rfloor} = 0.9999... 
\]</span> For <span class="math inline">\(K=14\)</span>: <span class="math display">\[
P = 1 - (1 - 0.98^{14})^{\left\lfloor\frac{50}{14}\right\rfloor} = 0.9850... 
\]</span></p>
</div>
</div>
</div>
<div id="section-3" class="section level4 unnumbered">
<h4></h4>
</div>
<div id="section-4" class="section level4 tabset">
<h4></h4>
<div id="question-1c" class="section level5">
<h5>Question 1C</h5>
<div class="question">
<p>Calculate the number of False Positives (FPs), i.e. the number of <span class="math inline">\(K\)</span>-mers that are expected to match by chance.</p>
</div>
</div>
<div id="formula-2" class="section level5">
<h5>Formula</h5>
<div class="answer">
<p><span class="math display">\[\begin{align*}
F &amp;= (Q-K+1) \times \bigg(\frac{G}{K}\bigg) \times \bigg(\frac{1}{A}\bigg)^{K},\\
&amp;A: \text{the alphabet size}\\
&amp;Q: \text{the query length in bases}\\
&amp;G: \text{the genome size in bases}
\end{align*}\]</span></p>
</div>
</div>
<div id="solution-2" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>For <span class="math inline">\(K=7\)</span>: <span class="math display">\[
F = (34-7+1) \times \frac{2500000000}{7} \times \frac{1}{4}^{7} = 610351.5625
\]</span> For <span class="math inline">\(K=14\)</span>: <span class="math display">\[
F = (34-14+1) \times \frac{2500000000}{14} \times \frac{1}{4}^{14} = 13.9698
\]</span></p>
</div>
</div>
</div>
<div id="section-5" class="section level4 unnumbered">
<h4></h4>
</div>
<div id="section-6" class="section level4 tabset">
<h4></h4>
<div id="question-1d" class="section level5">
<h5>Question 1D</h5>
<div class="question">
<p>Observe the True positive rate (TPR) and the number of False Positives for the <span class="math inline">\(7\)</span>-mers and <span class="math inline">\(14\)</span>-mers that you computed in part 1B and 1C. What observation do you make?</p>
</div>
</div>
<div id="solution-3" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>When increasing the <span class="math inline">\(K\)</span>-mer size both the TPR and the number of False Positives are reduced. But the number of False Positives reduces drastically, compared to the TPR.</p>
</div>
</div>
</div>
<div id="section-7" class="section level4 unnumbered">
<h4></h4>
</div>
</div>
<div id="exercise-2" class="section level1">
<h1>Exercise 2</h1>
<p>In order to increase the True Positive Rate, we want to allow single mismatches when checking for exact matches.</p>
<p>For the <span class="math inline">\(K\)</span>-mer sizes <span class="math inline">\(K=7\)</span> and <span class="math inline">\(K=14\)</span>, respectively.</p>
<div id="section-8" class="section level4 tabset">
<h4></h4>
<div id="question-2a" class="section level5">
<h5>Question 2A</h5>
<div class="question">
<p>What is the probability that at least one non-overlapping <span class="math inline">\(K\)</span>-mer in the homologous region matches perfectly with the corresponding <span class="math inline">\(K\)</span>-mer in the query sequence? Given that we allow one mismatch.</p>
</div>
</div>
<div id="formula-3" class="section level5">
<h5>Formula</h5>
<div class="answer">
<p><span class="math display">\[
P = 1 - (1 - (M^{K}+K\times M^{K-1}\times (1-M)))^{T},\\
\text{with}\quad T= \left\lfloor\frac{H}{K}\right\rfloor
\]</span></p>
</div>
</div>
<div id="solution-4" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>For <span class="math inline">\(K=7\)</span>: <span class="math display">\[
P = 1 - (1 - (0.98^{7} + 7*0.98^{6}*0.02))^{\left\lfloor\frac{50}{7}\right\rfloor} \approx 1
\]</span> For <span class="math inline">\(K=14\)</span>: <span class="math display">\[
P = 1 - (1 - (0.98^{14} + 14*0.98^{13}*0.02))^{\left\lfloor\frac{50}{14}\right\rfloor} \approx 1
\]</span></p>
</div>
</div>
</div>
<div id="section-9" class="section level4 unnumbered">
<h4></h4>
</div>
<div id="section-10" class="section level4 tabset">
<h4></h4>
<div id="question-2b" class="section level5">
<h5>Question 2B</h5>
<div class="question">
<p>Calculate the number of False Positives (FPs), i.e. the number of <span class="math inline">\(K\)</span>-mers that are expected to match by chance. Given that we allow one mismatch.</p>
</div>
</div>
<div id="formula-4" class="section level5">
<h5>Formula</h5>
<div class="answer">
<p><span class="math display">\[\begin{align*}
F &amp;= (Q-K+1) \times \bigg(\frac{G}{K}\bigg) \times \Bigg[\bigg(\frac{1}{A}\bigg)^{K} + K \times \bigg(\frac{1}{A}\bigg)^{K-1} \times \bigg(1-\frac{1}{A}\bigg)\Bigg],\\
&amp;A: \text{the alphabet size}\\
&amp;Q: \text{the query length in bases}\\
&amp;G: \text{the genome size in bases}
\end{align*}\]</span></p>
</div>
</div>
<div id="solution-5" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>For <span class="math inline">\(K=7\)</span>: <span class="math display">\[
F = (34-7+1) \times \frac{2500000000}{7} \times (\frac{1}{4}^{7} + 7 \times \frac{1}{4}^6 \times (1-\frac{1}{4})) = 1.3427... * 10^{7}  
\]</span> For <span class="math inline">\(K=14\)</span>: <span class="math display">\[
F = (34-14+1) \times \frac{2500000000}{14} \times (\frac{1}{4}^{14} + 14 \times \frac{1}{4}^{13} \times (1-\frac{1}{4})) = 600.7031  
\]</span></p>
</div>
</div>
</div>
<div id="section-11" class="section level4 unnumbered">
<h4></h4>
</div>
<div id="section-12" class="section level4 tabset">
<h4></h4>
<div id="question-2c" class="section level5">
<h5>Question 2C</h5>
<div class="question">
<p>What development do we see when observing the TPR and FPs for the updated formulae?</p>
</div>
</div>
<div id="solution-6" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>We observe an increase in both the TPR and the number of False positives.</p>
</div>
</div>
</div>
<div id="section-13" class="section level4 unnumbered">
<h4></h4>
</div>
</div>
<div id="exercise-3" class="section level1">
<h1>Exercise 3</h1>
<p>Finally, we want to reduce the number of False Positive results. To that end, instead of requiring single perfect matches, we now want at least <span class="math inline">\(n\)</span> perfect matches.</p>
<p>For the <span class="math inline">\(K\)</span>-mer sizes <span class="math inline">\(K=7\)</span> and <span class="math inline">\(K=14\)</span>, respectively.</p>
<div id="section-14" class="section level4 tabset">
<h4></h4>
<div id="question-3a" class="section level5">
<h5>Question 3A</h5>
<div class="question">
<p>What is the probability that at least <span class="math inline">\(2\)</span> non-overlapping <span class="math inline">\(K\)</span>-mer in the homologous region match perfectly with corresponding <span class="math inline">\(K\)</span>-mers in the query sequence?</p>
</div>
</div>
<div id="formula-5" class="section level5">
<h5>Formula</h5>
<div class="answer">
<p><span class="math display">\[
p_{n} = (M^{K})^{n} \times (1-M^{K})^{T-n} \times \frac{T!}{n! \times (T-n)!}\\
\text{with}\quad T= \left\lfloor\frac{H}{K}\right\rfloor\\\\
P = P_{n} + P_{n+1} + ... + P_{T}
\]</span></p>
</div>
</div>
<div id="solution-7" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>For <span class="math inline">\(K=7\)</span>: <span class="math display">\[
P_{2} = 0.98^{7 \times 2} \times (1-0.98^{7})^{\left\lfloor\frac{50}{7}\right\rfloor-2} \times \frac{\frac{50}{7}\text{!}}{2\text{!}\times(\frac{50}{7}-2)\text{!}}\\
\text{...}\\
\text{...}\\
\]</span> <span class="math display">\[\begin{align*}
P &amp;= P_{2} + P_{3} + P_{4} + P_{5} + P_{6} + P_{7}\\
  &amp;= 0.000631 + 0.006925 + 0.045591 + 0.180074 + 0.395142 + 0.371601\\
  &amp;= 0.999967...
\end{align*}\]</span></p>
<p>For <span class="math inline">\(K=14\)</span>: <span class="math display">\[
P_{2} = 0.98^{14 \times 2} \times (1-0.98^{14})^{\left\lfloor\frac{50}{14}\right\rfloor-2} \times \frac{\frac{50}{14}!}{2!\times(\frac{50}{14}-2)!}\\
\text{...}\\
\text{...}\\
\]</span> <span class="math display">\[\begin{align*}
P &amp;= P_{2} + P_{3}\\
  &amp;= 0.419776 + 0.428050\\
  &amp;= 0.847827...
\end{align*}\]</span></p>
</div>
</div>
</div>
<div id="section-15" class="section level4 unnumbered">
<h4></h4>
</div>
<div id="section-16" class="section level4 tabset">
<h4></h4>
<div id="question-3b" class="section level5">
<h5>Question 3B</h5>
<div class="question">
<p>We can observe a decrease in the TPR. Does it still make sense to use this method? Why?</p>
</div>
</div>
<div id="solution-8" class="section level5">
<h5>Solution</h5>
<div class="answer">
<p>While it is true that this also decreases the TPR. It drastically decreases the amount of False Positives, which greatly improves the overall results. In real examples, the size of the homologous regions is typically higher than 50 nucleotides that we chose for this example. When considering our example with <span class="math inline">\(H=100\)</span>, we would end up with a TPR close to 1.0, while having nearly no False Positives.</p>
</div>
</div>
</div>
<div id="section-17" class="section level4 unnumbered">
<h4></h4>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
